### ISSUE TITLE
Simplify of small imaginary number yields 0

### ISSUE THREAD

**Issue by TheLartians** [sympy/sympy#9398]
Using the most recent release (0.7.6) the following expression yields 0:  

    from sympy import *
    (I*Number(1.)*Number(10)**Number(-14)).simplify()

Using the absolute value this does not happen.

     (Number(1.)*Number(10)**Number(-14)).simplify()

**Comment by Sumith1896** [comment]
On master 
```
In [7]: (I*Number(1)*Number(10)**Number(-14)).simplify()
Out[7]: I/100000000000000

In [8]: (Number(1)*Number(10)**Number(-14)).simplify()
Out[8]: 1/100000000000000

In [9]: (I*Number(1.0)*Number(10)**Number(-14)).simplify()
Out[9]: 0

In [10]: (Number(1.0)*Number(10)**Number(-14)).simplify()
Out[10]: 1.00000000000000e-14
```
Doesn't seem to happen with ints, but happens with float.

**Comment by certik** [comment]
The problem is with simplifying floating points, i.e. with master (d9b59fd03c2a65b9dd783aa0103a98b88121061e):
```
In [1]: simplify(1e-14*I)
Out[1]: 0

In [2]: simplify(1e-14)
Out[2]: 1.00000000000000e-14
```
Here [2] is correct, but [1] is wrong.

**Comment by certik** [comment]
The problem is on the line https://github.com/sympy/sympy/blob/d9b59fd03c2a65b9dd783aa0103a98b88121061e/sympy/simplify/simplify.py#L3698, due to the fact that:
```
In [1]: cancel(1e-14*I)
Out[1]: 0

In [2]: cancel(1e-14)
Out[2]: 1.00000000000000e-14
```
So simplify() then sees 0 as one of the options and since it is short, it returns it. We need to fix `cancel()`.

**Comment by certik** [comment]
Found the problem. Mpmath converts 1e-14 to 0 here:
```
> /home/ondrej/.hashdist/bld/profile/rt2qwfzuuaxa/lib/python2.7/site-packages/mpmath/ctx_mp_python.py(77)__new__()
     76             v = new(cls)
---> 77             v._mpf_ = mpf_pos(cls.mpf_convert_arg(val, prec, rounding), prec, rounding)
     78             return v
```
Where:
```
ipdb> p val
1.00000000000000e-14
ipdb> p prec
53
ipdb> p v
0.0
```
Which might be correct. This is called from:
```
> /home/ondrej/repos/sympy/sympy/polys/domains/realfield.py(72)from_sympy()
     71         if number.is_Number:
---> 72             return self.dtype(number)
```
where `number=1e-14` and it returns `0`.  The issue is that polynomials convert the floating point number to a rational.

**Comment by certik** [comment]
So a simple workaround is this:
```diff
diff --git a/sympy/polys/domains/realfield.py b/sympy/polys/domains/realfield.py
index a34aa81..86aa367 100644
--- a/sympy/polys/domains/realfield.py
+++ b/sympy/polys/domains/realfield.py
@@ -25,7 +25,7 @@ class RealField(Field, CharacteristicZero, SimpleDomain):
     has_assoc_Ring = False
     has_assoc_Field = True
 
-    _default_precision = 53
+    _default_precision = 100
 
     @property
     def has_default_precision(self):
```
Now it works:
```
In [1]: simplify(1e-14*I)
Out[1]: 1.0e-14⋅ⅈ
```
as well as the original issue:
```
In [2]: (I*Number(1.0)*Number(10)**Number(-14)).simplify()
Out[2]: 1.0e-14⋅ⅈ
```

The patch just increases the precision of the `RR` ring. Now we just need to figure out how to do this automatically somehow, perhaps based on the accuracy of the numbers that enter `simplify()`.

**Comment by certik** [comment]
I posted a possible fix in #9403.

Do you think this is the right fix for this problem?
